frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -248,27 +248,95 @@ function ReaderHighlight:onTapXPointerSavedHighlight(ges)
+            if updated_highlight_beginning then
+                self.view.highlight.saved[page][index].pos0 = updated_highlight_beginning
+            end
+        end
+    else -- we move from pos1
+        if direction == 1 then
+            local updated_highlight_end = self.ui.document:getNextVisibleWordEnd(highlight_end)
+            if updated_highlight_end then
+                self.view.highlight.saved[page][index].pos1 = updated_highlight_end
+            end
+        else
+            local updated_highlight_end = self.ui.document:getPrevVisibleWordEnd(highlight_end)
+            if updated_highlight_end then
+                self.view.highlight.saved[page][index].pos1 = updated_highlight_end
+            end
+        end
+    end
+    UIManager:setDirty(self.dialog, "ui")
+end
+
 function ReaderHighlight:onShowHighlightDialog(page, index)
-    self.edit_highlight_dialog = ButtonDialog:new{
-        buttons = {
-            {
+    local buttons = {
                 {
-                    text = _("Delete"),
-                    callback = function()
-                        self:deleteHighlight(page, index)
-                        -- other part outside of the dialog may be dirty
-                        UIManager:close(self.edit_highlight_dialog, "ui")
-                    end,
-                },
-                {
-                    text = _("Edit"),
-                    callback = function()
-                        self:editHighlight(page, index)
-                        UIManager:close(self.edit_highlight_dialog)
-                    end,
+                    {
+                        text = _("Delete"),
+                        callback = function()
+                            self:deleteHighlight(page, index)
+                            -- other part outside of the dialog may be dirty
+                            UIManager:close(self.edit_highlight_dialog, "ui")
+                        end,
+                    },
+                    {
+                        text = _("Edit"),
+                        callback = function()
+                            self:editHighlight(page, index)
+                            UIManager:close(self.edit_highlight_dialog)
+                        end,
+                    },
                 },
-            },
-        },
+             }
+
+    if not self.ui.document.info.has_pages then
+        table.insert(buttons, {
+                        {
+                            text = _("◁⇱"),
+                            callback = function()
+                                self:updateHighlight(page, index, 0, -1)
+                            end,
+                        },
+                        {
+                            text = _("⇱▷"),
+                            callback = function()
+                                self:updateHighlight(page, index, 0, 1)
+                            end,
+                        },
+                    })
```

Comment:
Small thing. Can you shift the whole block 12 spaces left? The closing `})` shoud vertically align with the start of `table.insert(buttons, {`.
Also the one with Delete|Edit, that does not need all that indentation now that you made it less nested.

Reply:
Sure, I'll do it once I get my emulator rebuild